src/com/google/javascript/jscomp/CodeGenerator.java

```diff
@@ -1061,7 +1061,7 @@ protected void add(Node n, Context context) {
           }
         }
         add("[");
-        add(first);
+        addExpr(first, 1, Context.OTHER);
```

Comment:
Could you add a comment both here?

```java
// Must use addExpr() with a priority of 1, because comma expressions aren't allowed.
// LINK TO SPEC HERE
```